spec.bs

```diff
@@ -2890,9 +2894,9 @@ content in the <{fencedframe}> or its embedder, exactly one of two things will h
 
 <div algorithm=create-and-initialize-document-patch>
   Modify [[!HTML]]'s [=create and initialize a Document object=] algorithm to insert one step after
-  step 2 that reads:
+  step 3 that reads:
 
-  3. If |navigationParams|'s [=navigation params/fenced frame config instance=] is not null:
+  4. If |navigationParams|'s [=navigation params/fenced frame config instance=] is not null:
```

Comment:
Actually I'm not 100% clear on how the `|permissionsPolicy|` returned from https://html.spec.whatwg.org/multipage/document-lifecycle.html#shared-document-creation-infrastructure%3Anav-bc:~:text=Let%20permissionsPolicy%20be%20the%20result%20of%20creating%20a%20permissions%20policy%20from%20a%20response%20given%20navigationParams%27s%20navigable%27s%20container%2C%20navigationParams%27s%20origin%2C%20and%20navigationParams%27s%20response.%20%5BPERMISSIONSPOLICY%5D is deficient. Can you describe exactly where in https://w3c.github.io/webappsec-permissions-policy/#create-from-response or https://w3c.github.io/webappsec-permissions-policy/#create-for-navigable we're going wrong, and why we need to amend it?

Reply:
The existing spec looks at [the frame's](https://w3c.github.io/webappsec-permissions-policy/#ref-for-parse-policy-directive:~:text=document%2C%20and-,element%E2%80%99s%20declared%20origin.,-If%20element%E2%80%99s) [declared origin](https://w3c.github.io/webappsec-permissions-policy/#declared-origin) to determine [what the 'src' allowlist should map to](https://w3c.github.io/webappsec-permissions-policy/#parse-policy-directive:~:text=%27src%27%22%3A-,Let%20allowlist%E2%80%99s%20src%2Dorigin%20be%20target%20origin.,-Continue%20to%20the) when building the policy. This algorithm has no concept of `config` or mapped URLs. For fenced frames, since there is no 'src' attribute, it won't be able to pull whatever is set in the 'src' attribute, and will instead pull [whatever the document is currently navigated to](https://w3c.github.io/webappsec-permissions-policy/#declared-origin:~:text=%E2%80%99s%20origin.-,Return%20node%E2%80%99s%20node%20document%E2%80%99s%20origin.,-The%20declared%20origin), including for subsequent navigations. For URN iframes, it will just pull the `urn:uuid` that the `src` attribute is set to.

When I first looked at this, I had thought that [declared origin](https://w3c.github.io/webappsec-permissions-policy/#declared-origin) was called well before navigation commit (this is how it is handled in the Chromium implementation). Now that I'm taking another look at this, it looks like the permissions policy is constructed (and declared origin is called) during the [initialize the document object](https://html.spec.whatwg.org/#initialise-the-document-object) algorithm, so I think we can instead modify [declared origin](https://w3c.github.io/webappsec-permissions-policy/#declared-origin) directly to take installed fenced frame config instances into account, since that will be installed and accessible by that point.

I will need to move the "If |navigationParams|'s [=navigation params/fenced frame config instance=] is not null" step from step 4 back to step 3 so that the fenced frame config instance is in place before the permissions policy is built.